### ISSUE TITLE
[ISSUE] Issue with `databricks_grants` resource

### ISSUE THREAD

**Issue by sveerella** [databricks/terraform-provider-databricks#3407]
We're creating multiple external locations in terraform loop and also the grants for it. We're attaching different kind of grants for two different groups. We observe that these grants are getting missed adding to the few external locations. We see that the response from the terraform run says the resource is successfully created though/

### Configuration

```hcl

variable "s3_bucket_paths" {
  description = "Directory paths of RMS S3 which tenant needs to have access to"
  type    = list(string)
  default = ["dir1", "dir2", "dir3" ]
}

resource "databricks_external_location" "data" {
  count = length(var.s3_bucket_paths)
  force_destroy = "true"
  lifecycle { 
      ignore_changes = all
  }
  provider = databricks.created_workspace
  name            = "${var.prefix}-${var.tenant_id}-${count.index + 1}"
  url             = "s3://${var.s3_bucket_name}/${var.s3_bucket_paths[count.index]}"
  credential_name = "${var.prefix}-${var.tenant_id}"
}


resource "databricks_grants" "external_location_access" {
  count = length(var.s3_bucket_paths)
  provider = databricks.created_workspace
  external_location = databricks_external_location.data[count.index].id
  lifecycle { 
      ignore_changes = all
  }
  grant {
    principal  = databricks_group.test_eng.display_name
    privileges = ["ALL_PRIVILEGES"]
  }
  grant {
    principal  = databricks_group.data_eng.display_name
    privileges = ["READ_FILES"]
  }
}
```

### Expected Behavior
With the above configuration we would expect all grants to be added to both the groups in all external locations

### Actual Behavior
But we see that the grants aren't getting added to few external locations, even though terraform run says `creation complete`.
![Screenshot 2024-03-27 at 11 28 33 AM](https://github.com/databricks/terraform-provider-databricks/assets/102265965/ceca2ecc-81c9-4399-9632-11984dc34bdd)

we see that it's getting added to few external location, for example check the screenshot below
![Screenshot 2024-03-27 at 11 24 55 AM](https://github.com/databricks/terraform-provider-databricks/assets/102265965/9135bd82-3d6c-4daa-86bc-1fa87f6951f5)

for few other external location, grants are added to ONLY one group like in screenshot below
![Screenshot 2024-03-27 at 11 30 26 AM](https://github.com/databricks/terraform-provider-databricks/assets/102265965/67649a2e-f145-4bc0-823b-a13ddd91d677)

### Steps to Reproduce
1. Create multiple external locations in a loop using the terraform code above

### Terraform and provider versions
```
terraform version
Terraform v1.4.6
on darwin_amd64
+ provider registry.terraform.io/databricks/databricks v1.35.0
+ provider registry.terraform.io/hashicorp/aws v4.58.0
+ provider registry.terraform.io/hashicorp/random v3.6.0
+ provider registry.terraform.io/hashicorp/time v0.11.1

Your version of Terraform is out of date! The latest version
is 1.7.5. You can update by downloading from https://www.terraform.io/downloads.html

```

### Is it a regression?
no

**Comment by nkvuong** [comment]
@sveerella what happens when you re-run terraform plan? I notice that you are specifying `lifecycle {ignore_changes = all}`, which blocks any future drift detection

**Comment by sveerella** [comment]
Yes, I added that because the privileges keeps on updating for every terraform run.
for example if all the grants are created in first run, then in second run its getting deleted and in next run they're getting added. To avoid this behavior we added the `lifecycle {ignore_changes = all}, ` and want to add all the grants in first run.
